jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/TaskCountLoadCalculator.java

```diff
@@ -0,0 +1,96 @@
+package org.jbpm.services.task.assignment.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.jbpm.services.task.assignment.UserTaskLoad;
+import org.jbpm.services.task.utils.ClassUtil;
+import org.kie.api.task.TaskContext;
+import org.kie.api.task.model.User;
+import org.kie.internal.task.api.TaskPersistenceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TaskCountLoadCalculator extends LoadCalculatorImpl {
+	private static final String IDENTIFIER = "TaskCountLoadCalculator";
+	private static final Logger logger = LoggerFactory.getLogger(TaskCountLoadCalculator.class);
+    private static final String SINGLE_USER_QUERY = 
+            "select new Integer(count(t)) "
+            + "from TaskImpl t "
+            + "where t.taskData.actualOwner.id = :owner and t.taskData.status in ('Reserved', 'InProgress', 'Suspended') ";
+    private static final String MULTI_USER_QUERY = 
+            "select new org.jbpm.services.task.assignment.impl.AssignmentImpl(t.taskData.actualOwner.id, count(t)) "
+            + "from TaskImpl t "
+            + "where t.taskData.actualOwner.id in (:owners) and t.taskData.status in ('Reserved', 'InProgress', 'Suspended') "
+            + "group by t.taskData.actualOwner "
+            + "order by count(t) asc, t.taskData.actualOwner.id asc";
```

Comment:
what's the point of having these two queries? Can't we only use the MULTI_USEER_QUERY and in case of single user just give a list with one user in it? Anyway the list is processed in both cases so we could reuse bit of code